der/src/document.rs

```diff
@@ -1,67 +1,102 @@
 //! ASN.1 DER-encoded documents stored on the heap.
 
 use crate::{Decodable, Encodable, Error, Result};
-use alloc::{boxed::Box, vec::Vec};
+use alloc::vec::Vec;
+use core::marker::PhantomData;
 
 #[cfg(feature = "pem")]
-use {crate::pem, alloc::string::String};
+use {
+    crate::pem::{self, PemLabel},
+    alloc::string::String,
+};
 
 #[cfg(feature = "std")]
 use std::{fs, path::Path};
 
+/// Marker trait which identifies documents which contain sensitive
+/// information, such as private keys.
+pub trait Sensitivity {
+    /// Does this type contain potentially sensitive data?
+    ///
+    /// This enables hardened file permissions when persisting data to disk.
+    const SENSITIVE: bool;
+}
```

Comment:
Marker trait for types which represent private keys. The effects should probably be better documented in rustdoc, but basically come down to:

1. Zeroizing `der_bytes` when the `zeroize` crate feature is enabled
2. Setting appropriate file permissions when writing DER or PEM data to disk

Alternatively this could be a const generic parameter. It's something to consider, but exposing the information under a trait might be useful in other settings.

Reply:
If we gave this trait another name, we could also include the PEM "type" as a `&'static str` here too.